File: 1701-average-waiting-time/1701-average-waiting-time.js
var averageWaitingTime = function(customers) {
    let totalTimePassed = 0;
    let waitTime = 0;

    for (const customer of customers) {
        const arrivalTime = customer[0];
        const serviceTime = customer[1];
        
        totalTimePassed = Math.max(arrivalTime, totalTimePassed); 
        totalTimePassed += serviceTime; 
        waitTime += (totalTimePassed - arrivalTime); 
    }

    return waitTime / customers.length; 
};

File: 1701-average-waiting-time/1701-average-waiting-time.java
class Solution {
    public double averageWaitingTime(int[][] customers) {
        long totalTimePassed = 0;
        long totalWaitTime = 0;
        
        for(int i = 0; i < customers.length; i++) {
            totalTimePassed = Math.max(totalTimePassed, customers[i][0]) + customers[i][1];
            totalWaitTime += totalTimePassed - customers[i][0];
           
        }
        
        double result = (double)totalWaitTime / customers.length;
        return result;
    }
}